app/src/main/java/org/greenstand/android/TreeTracker/signup/NameEntryView.kt

```diff
@@ -117,7 +120,9 @@ fun NameEntryView(
                 ),
                 keyboardActions = KeyboardActions(
                     onGo = {
-                        cameraLauncher.launch(true)
+                        if (state.name != null && state.organization != null) {
+                            cameraLauncher.launch(true)
+                        }
```

Comment:
-nit-

We can try to remove all business logic from this view by making this if statement logic into an extension function on the state object. The extension function can contain that boolean logic.

if (state.canGoToNextScreen) {
   // go to next screen
}

What are your thoughts?

Reply:
Yes this would be much better but I would suspend this for now. Gwen made some note about validations approach so the implementation would change soon.

Reply:
We can put it in another PR if you'd like. But it should be done. The 2.0 milestone for the app will have the functionality of the old one, which validates email and phone number.

Once we move on to 2.1+ we'll start adding in new features such as org flavors, messages screen, etc..